spec/models/document_spec.rb

```diff
@@ -274,4 +274,119 @@
       end
     end
   end
+
+  describe "#update_or_replace" do
+    let(:planning_application) { create(:planning_application, :not_started) }
+
+    let(:file_path1) do
+      Rails.root.join("spec/fixtures/images/proposed-floorplan.png")
+    end
+
+    let(:file_path2) do
+      Rails.root.join("spec/fixtures/images/proposed-roofplan.png")
+    end
+
+    let(:file1) { Rack::Test::UploadedFile.new(file_path1, "image/png") }
+    let(:file2) { Rack::Test::UploadedFile.new(file_path2, "image/png") }
+
+    let(:document) do
+      create(
+        :document,
+        file: file1,
+        planning_application: planning_application,
+        numbers: "DOC123"
+      )
+    end
+
+    context "when there is no 'file' attribute" do
```

Comment:
This is just a general question about contexts and nothing to do with this PR.

In general, contexts cover the set of data so they are often paired... so in this case you have "when there is **no** 'file' attribute" and when there is a 'file' attribute??

It's just that I've been thinking about some of the contexts I've seen in other PRs.

Reply:
Hmm. I've been vaguely wondering whether Unboxed should have a 'handbook' with style guides etc. This is definitely the sort of thing that should be considered in it!

I think the way I set up contexts is probably a bit haphazard. More `context: "I needed a slightly different data set up"` than anything else 😅

Reply:
I know my test style is different from the rest of the team so I am trying to make sense of the team style.

I don't think there's anything wrong with what you've done here. But, when I tried to imitate the style it seems to be the tail wagging the dog, I need variation in the test setup so I need another context but I don't have any good reasoning for it. The only way around this is to increase the nesting which isn't idea either.

Anyway thanks.